Bug: [no-unused-vars] 'Component' is defined but never used

### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

does not work

### Repro Code

```typescript
import { Component } from '@angular/core'

@Component({
  selector: 'app-root',
  template: '',
})
export class AppComponent {}
```


### ESLint Config

```javascript
{
  "root": true,
  "overrides": [
    // ....
    {
      "files": ["*.ts"],
      "parser": "@typescript-eslint/parser",
      "plugins": ["tsdoc", "@typescript-eslint"],
      "extends": [
        "plugin:@typescript-eslint/recommended",
        "plugin:@typescript-eslint/recommended-requiring-type-checking",
        "plugin:@typescript-eslint/strict"
      ],
      "parserOptions": {
        "project": ["tsconfig.lint.json"]
      },
      "rules": {
        "tsdoc/syntax": "error"
      }
    },
    {
      "files": ["src/**/*.ts"],
      "env": {
        "node": false,
        "browser": true
      },
      "extends": [
        "plugin:compat/recommended",
        "plugin:@angular-eslint/recommended",
        "plugin:@angular-eslint/template/process-inline-templates"
      ],
      "parserOptions": {
        "project": ["tsconfig.lint.json"]
      },
      "rules": {
        "no-alert": "error",
        "no-debugger": "error",
        "no-console": "error",
        "@angular-eslint/component-class-suffix": [
          "error",
          {
            "suffixes": ["Page", "Component"]
          }
        ],
        "@typescript-eslint/explicit-module-boundary-types": [
          "off",
          {
            "allowHigherOrderFunctions": true
          }
        ]
      }
    },
    {
      "files": ["*.js", "*.mjs", "*.ts", "*.mts", "*.html"],
      "extends": ["prettier"]
    }
  ]
}
```


### tsconfig

```jsonc
{
  "$schema": "http://json.schemastore.org/tsconfig",
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "./dist/out-tsc",
    "strict": true,
    "noImplicitAny": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitOverride": true,
    // "noPropertyAccessFromIndexSignature": true,
    "noFallthroughCasesInSwitch": true,
    "forceConsistentCasingInFileNames": true,
    "strictPropertyInitialization": true,
    "strictNullChecks": true,
    "esModuleInterop": true,
    "downlevelIteration": true,
    "sourceMap": true,
    "declaration": false,
    "experimentalDecorators": true,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "moduleResolution": "node",
    "importHelpers": true,
    "target": "ES2020",
    "module": "ESNext",
    "lib": ["ESNext", "DOM"]
  },
  "angularCompilerOptions": {
    "fullTemplateTypeCheck": true,
    "enableI18nLegacyMessageIdFormat": false,
    "strictInjectionParameters": true,
    "strictInputAccessModifiers": true,
    "strictTemplates": true
  }
}
```


### Expected Result

Recognise imported "Component" as "experimentalDecorators".

### Actual Result

'Component' is defined but never used.eslint[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)

### Additional Info

_No response_

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `^5.36.0` |
| `@typescript-eslint/parser`            | `^5.36.0` |
| `TypeScript`                           | `<4.8.0` |
| `ESLint`                               | `^8.23.0` |
| `node`                                 | `17` |

I have the similar issue. Here is my example:
A.ts:
```
export interface A {
  name: string;
}
```
B.ts:
```
import { A } from './A';

export class B {
  a?: A;
  static {}
}
```
@typescript-eslint/eslint-plugin: ^5.36.0
@typescript-eslint/parser: ^5.36.0
The error is:
```
  1:10  error  'A' is defined but never used  @typescript-eslint/no-unused-vars
```
If I remove the static block, then everything works correctly.

--- a/package.json
+++ b/package.json
@@ -108,7 +108,7 @@
     "typescript": ">=3.3.1 <4.9.0"
   },
   "resolutions": {
-    "typescript": "4.8.2",
+    "typescript": "~4.8.2",
     "@types/node": "^17.0.31",
     "pretty-format": "^28.1.0",
     "//": "Pin jest to v28 across the repo",

--- a/packages/typescript-estree/src/convert.ts
+++ b/packages/typescript-estree/src/convert.ts
@@ -1056,15 +1056,11 @@ export class Converter {
 
         /**
          * Semantically, decorators are not allowed on variable declarations,
-         * but the TypeScript compiler will parse them and produce a valid AST,
-         * so we handle them here too.
+         * Pre 4.8 TS would include them in the AST, so we did as well.
+         * However as of 4.8 TS no longer includes it (as it is, well, invalid).
+         *
+         * So for consistency across versions, we no longer include it either.
          */
-        const decorators = getDecorators(node);
-        if (decorators) {
-          (result as any).decorators = decorators.map(el =>
-            this.convertChild(el),
-          );
-        }
 
         if (hasModifier(SyntaxKind.DeclareKeyword, node)) {
           result.declare = true;

--- a/packages/typescript-estree/src/getModifiers.ts
+++ b/packages/typescript-estree/src/getModifiers.ts
@@ -47,6 +47,6 @@ export function getDecorators(
 
   return (
     // eslint-disable-next-line deprecation/deprecation -- intentional fallback for older TS versions
-    node.modifiers?.filter(ts.isDecorator)
+    node.decorators?.filter(ts.isDecorator)
   );
 }

--- a/patches/typescript+4.8.2.patch
+++ b/patches/typescript+4.8.2.patch
@@ -1,5 +1,5 @@
 diff --git a/node_modules/typescript/lib/typescript.d.ts b/node_modules/typescript/lib/typescript.d.ts
-index 0fd60ae..b3610b8 100644
+index 0fd60ae..8bce90e 100644
 --- a/node_modules/typescript/lib/typescript.d.ts
 +++ b/node_modules/typescript/lib/typescript.d.ts
 @@ -425,8 +425,8 @@ declare namespace ts {
@@ -40,3 +40,12 @@ index 0fd60ae..b3610b8 100644
      function canHaveDecorators(node: Node): node is HasDecorators;
  }
  declare namespace ts {
+@@ -7924,7 +7936,7 @@ declare namespace ts {
+          * const decorators = ts.canHaveDecorators(node) ? ts.getDecorators(node) : undefined;
+          * ```
+          */
+-        readonly decorators?: undefined;
++        readonly decorators?: NodeArray<Decorator> | undefined;
+         /**
+          * @deprecated `modifiers` has been removed from `Node` and moved to the `Node` subtypes that support them.
+          * Use `ts.canHaveModifiers()` to test whether a `Node` can have modifiers.

